pkg/controller/cloneset/sync/cloneset_scale.go

```diff
@@ -278,6 +290,17 @@ func (r *realControl) deletePods(cs *appsv1alpha1.CloneSet, podsToDelete []*v1.P
 				modified = true
 				clonesetutils.ResourceVersionExpectations.Expect(gotPod)
 			}
+
+			if lifecycle.IsPodNotReadyDuringPreparingDeletePolicy(cs.Spec.Lifecycle) {
+				if updated, err := podreadiness.SetPodNotReadyDuringPreparingDelete(r.Client, pod, getReadinessMessage(cs)); err != nil {
+					return false, err
```

Comment:
1. If it fails, the pod in PreparingDelete state will still be ready. So maybe add an error log to warn this.
2. Should this `SetPodNotReady` logic be before or after the `UpdatePodLifecycle` logic? Should we rollback the first operation if the second failed?